components/Modal.tsx

```diff
@@ -0,0 +1,58 @@
+import DistrictData from '../data/districts.json';
+
+type props ={
+  pageX: number;
+  pageY: number;
+  id: string;
+  isEmpty?: boolean;
+}
+
+type districtDetails = {
+  label: string;
+  value: string | number;
+}
+
+
+const Modal = (props: props) => {
+  const { pageY, pageX, id } = props || {}
+  const district = DistrictData[parseInt(id)]
+
+  const districtDetails: districtDetails[] = [
+    {
+      label: "Elevation",
+      value: district?.elevation
+    },
+    {
+      label: "Area",
+      value: `${district?.area.toLocaleString('en-Us')} km²`
+    },
+    {
+      label: "Province",
+      value: district?.province
+    },
+    {
+      label: "Population",
+      value: district?.population
```

Comment:
Since, elevation area, province and populations are always present on the data, optional chaining isn't required

Reply:
OK I had this for previous version and forgot to remove